components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java

```diff
@@ -63,9 +63,9 @@ private APIGatewayManager() {
         this.artifactSaver = ServiceReferenceHolder.getInstance().getArtifactSaver();
     }
 
-    public synchronized static APIGatewayManager getInstance() {
+    public synchronized static APIGatewayManager getInstance(String organization) {
         if (instance == null) {
-            instance = new APIGatewayManager();
+            instance = new APIGatewayManager(organization);
         }
         return instance;
```

Comment:
This will break if we call this second time with a different organization. Instance is already initialized with the previous organization.
It looks like the issue is there in the previous code as well..

Reply:
looks like even in the previous code, the **environments** map hasn't been used anywhere after initializing.